dlrover/trainer/torch/flash_checkpoint/ckpt_backup.py

```diff
@@ -127,13 +119,16 @@ def _gather_peer_ckpt(self, buffer, meta_data):
         # Allgather tensor sizes
         dist.all_gather(shm_size_list, local_size, group=self._backup_group)
 
-        output_tensors = []
-        for tensor_size in shm_size_list:
-            output_tensor = torch.empty(
-                tensor_size, dtype=torch.uint8, device=self.current_device
-            )
-            output_tensors.append(output_tensor)
+        max_tensor_size = int(max(shm_size_list))
+        # Resize tensor to max size across all ranks.
+        byte_tensor.resize_(max_tensor_size)
 
+        output_tensors = [
+            torch.empty(
```

Comment:
The formatting here is strange. Better to manually adjust to:
```
for _ in range(x):
   xxx
```

Reply:
![image](https://github.com/intelligent-machine-learning/dlrover/assets/18071380/2bedbe82-8c5d-4d65-afd8-87a663d50850)